pkg/agent/controller/networkpolicy/networkpolicy_controller.go

```diff
@@ -94,6 +97,16 @@ func NewNetworkPolicyController(antreaClientGetter agent.AntreaClientProvider,
 	// Wait until appliedToGroupWatcher, addressGroupWatcher and networkPolicyWatcher to receive bookmark event.
 	c.fullSyncGroup.Add(3)
 
+	if c.ofClient != nil && c.ofClient.GetAntreaPolicyConfig() {
```

Comment:
Can we pass the feature gate value to NewNetworkPolicyController() instead of getting that from ofClient?